Allow to use different icon sets in the backend

javiereguiluz opened the pull request:
**UPDATE**: I changed the feature behavior completely since the first implementation. We no longer include the entire Tabler icon set (it took more than 22 MB of disk space) but the feature is still flexible and allows you to use any custom icon set, including tabler.

-----

This feature is the first that uses a Twig Component. I hope to add many other components soon.

Thanks @yceruto for publishing a fantastic article about how to use Twig Components in bundles: https://dev.to/yceruto/bundling-your-symfony-ux-twig-components-4997

yceruto commented:
>Thanks @yceruto for publishing a fantastic article about how to use Twig Components in bundles

You're very welcome! I'm glad you enjoyed the article 😊 you were my inspiration after all

alexislefebvre commented:
This PR adds 2300+ SVG files, will it slow down the installation of this bundle? I'm thinking about all these files which will be decompressed and created on every CI job that use EasyAdmin.

I don't have an alternative solution to offer.

javiereguiluz commented:
@alexislefebvre yes, I thought about that.

Initially, this PR also added Tabler icons, which added +6,000 additional icons. I removed that because it was too much. But, in the case of FontAwesome, I think it's a reasonable tradeoff. This adds a ~2,000 small files, but also removes +2,000 CSS classes injected in all admin pages. We can also validate icon names, trigger deprecations when legacy icon names are used, etc. So, I think it's worth it 🙌

pavol-tuka commented:
This PR made SlugField's javascript for lock/unlock not working because of this:
```
this.lockButtonIcon = this.lockButton.querySelector('i');
```
https://github.com/EasyCorp/EasyAdminBundle/blob/4.x/assets/js/field-slug.js#L86

There is no `<i>` element now.

javiereguiluz commented:
@pavol-tuka good catch! Fixed in #6622. Thanks.
